File: benchmarks/tsb/bench_period_arithmetic.ts
/**
 * Benchmark: Period.add / diff / compareTo / contains — Period arithmetic on 1k periods.
 * Outputs JSON: {"function": "period_arithmetic", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Period } from "../../src/index.ts";

const SIZE = 1_000;
const WARMUP = 5;
const ITERATIONS = 50;

const base = Period.fromDate(new Date(Date.UTC(2020, 0, 1)), "D");
const periods = Array.from({ length: SIZE }, (_, i) => base.add(i));
const other = base.add(500);

for (let i = 0; i < WARMUP; i++) {
  for (const p of periods.slice(0, 50)) {
    p.add(10);
    p.diff(other);
    p.compareTo(other);
    p.contains(p.startTime);
  }
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  for (const p of periods) {
    p.add(10);
    p.diff(other);
    p.compareTo(other);
    p.contains(p.startTime);
  }
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "period_arithmetic",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_period_arithmetic.py
"""Benchmark: Period.add / diff / compareTo / contains — Period arithmetic on 1k periods."""
import json, time
import pandas as pd

SIZE = 1_000
WARMUP = 5
ITERATIONS = 50

base = pd.Period("2020-01-01", freq="D")
periods = [base + i for i in range(SIZE)]
other = base + 500

for _ in range(WARMUP):
    for p in periods[:50]:
        p + 10
        p - other
        p < other
        p.start_time

times = []
for _ in range(ITERATIONS):
    t0 = time.perf_counter()
    for p in periods:
        p + 10
        p - other
        p < other
        p.start_time
    times.append((time.perf_counter() - t0) * 1000)

total_ms = sum(times)
print(json.dumps({"function": "period_arithmetic", "mean_ms": round(total_ms / ITERATIONS, 3), "iterations": ITERATIONS, "total_ms": round(total_ms, 3)}))
